### PR TITLE
fix(gui): enable initial window tab docking

### PR DESCRIPTION
**Pull request by akiojin**
## Summary

- Enables initial window tab docking from an ungrouped window by using titlebar drag as the entry point.
- Restricts docking to another window titlebar so body/canvas drops continue to move the window.
- Adds behavior-level frontend coverage for titlebar hit-testing and keeps SPEC-2008 Phase 14 manual smoke open.

## Changes

- `crates/gwt/web/window-docking.js`: Adds a pure titlebar hit-test helper with z-index priority, source exclusion, hidden-tab exclusion, and body-drop rejection.
- `crates/gwt/web/app.js`: Wires titlebar drag preview and sends `dock_window_tab` only when pointerup lands on another titlebar.
- `crates/gwt/web/index.html`: Adds dock target titlebar preview styling.
- `crates/gwt/src/embedded_web.rs` / `crates/gwt/src/embedded_server.rs`: Embeds and serves the new `window-docking.js` module.
- `crates/gwt/web/__tests__` / `package.json`: Adds docking behavior coverage and includes it in frontend unit/bundle scripts.
- `tasks/lessons.md`: Records the interaction-test lesson from this regression.

## Testing

- [x] `pnpm test:frontend-bundle` - passed.
- [x] `pnpm test:frontend-unit` - passed, 237 tests.
- [x] `cargo test -p gwt-core -p gwt` - passed.
- [x] `cargo clippy --all-targets --all-features -- -D warnings` - passed.
- [x] `cargo fmt -- --check` - passed.
- [x] `cargo build -p gwt` - passed.
- [x] Embedded URL Playwright chrome/reduced-motion subset - passed, 6 tests.
- [ ] Manual GUI docking smoke for Agent/Board/Issue dock, tab switch, detach, close, restart restore - not run in this session; tracked as SPEC-2008 T-121.

## Closing Issues

- None

## Related Issues / Links

- #2008

## Checklist

- [x] Tests added/updated
- [x] Lint/format passed (`cargo clippy`, `cargo fmt`)
- [ ] Documentation updated (if user-facing change) - Not updated; SPEC-2008 artifacts were updated and this is an interaction fix for an existing documented feature.
- [x] Migration/backfill plan included (if schema/data change) - No schema/data migration required.
- [x] CHANGELOG impact considered (breaking change flagged in commit) - Patch fix, no breaking change.

## Context

SPEC-2008 Phase 14 already had backend/protocol/persistence support for dockable window tabs, but ungrouped windows hid the tab strip and therefore had no initial visible drag source. The accepted fix is titlebar drag as the initial dock entry.

## Risk / Impact

- **Affected areas**: Workspace window titlebar drag/drop, tab docking preview, embedded frontend asset routing.
- **Rollback plan**: Revert this PR; existing grouped tab-strip docking behavior returns to the previous implementation.

## Notes

- This PR is draft until SPEC-2008 T-121 manual GUI smoke is completed.
- Full `pnpm test:visual` with an embedded URL still fails in existing `theme-toggle.spec.ts` aria-checked assertions; the scoped chrome/reduced-motion smoke passed and the failure is not caused by this docking change.

## Summary by CodeRabbit

* **New Features**
  * Added window tab docking functionality allowing tabs to be docked to other visible windows with live visual preview feedback during drag operations.

* **Tests**
  * Added comprehensive test coverage for tab docking hit-testing, z-index prioritization, and edge cases (hidden tabs, source window exclusion).
  * Extended operator UI chrome structure test assertions.

* **Chores**
  * Updated frontend test scripts for expanded module coverage.
  * Added lessons learned documentation on window tab interaction verification practices.

### CODE REVIEW

File: crates/gwt/web/app.js [hunk 1]
```diff
@@ -7921,13 +7974,26 @@
...
-            sendGeometry(
-              dragState.id,
-              runtime?.terminal.cols || 80,
-              runtime?.terminal.rows || 24,
-            );
+            dragState.dockTargetId = dragState.allowMove
+              ? titlebarDockTargetAt(event, dragState.id)
+              : null;
+            clearTitlebarDockPreview();
+            if (dragState.dockTargetId) {
```
[the remark this answers is not in the document]
Comment (akiojin) [reply crates/gwt/web/app.js:8083]: Fixed in f4b57d28: findTitlebarDockTarget now returns no target when the source window is already in a tab group, so grouped window titlebar drags continue to move the group instead of redocking. Added frontend unit coverage for the grouped-source titlebar drag case.

File: crates/gwt/web/window-docking.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        const titlebarHeight = Math.min(hitHeight, Math.max(0, geometry.height));
+        return (
+          point.x >= geometry.x &&
+          point.x <= geometry.x + geometry.width &&
+          point.y >= geometry.y &&
+          point.y <= geometry.y + titlebarHeight
+        );
+      })?.id || null
+  );
+}
```
[the remark this answers is not in the document]
Comment (akiojin) [reply crates/gwt/web/window-docking.js:32]: Fixed in f4b57d28: findTitlebarDockTarget now returns no target when the source window is already in a tab group, so grouped window titlebar drags continue to move the group instead of redocking. Added frontend unit coverage for the grouped-source titlebar drag case.

File: tasks/lessons.md [hunk 3]
```diff
@@ -1,5 +1,36 @@
...
+
+1. Pointer / DnD / keyboard interaction を追加するときは、可能な限り
+   hit-test や state transition を小さな純粋関数に切り出し、`node --test`
+   で座標・負例・優先順位まで固定する。
+2. Source-string contract は配線漏れ検出に限定し、操作可能性の唯一の
+   根拠にしない。
+3. GUI smoke で embedded frontend を検証する場合は `cargo run -p gwt --bin gwt`
+   が出す `gwt browser URL` を使う。static web root 配信は embedded route
+   と同等ではない。
+
```
[the remark this answers is not in the document]
Comment (akiojin) [reply tasks/lessons.md:33]: Fixed in f4b57d28: findTitlebarDockTarget now returns no target when the source window is already in a tab group, so grouped window titlebar drags continue to move the group instead of redocking. Added frontend unit coverage for the grouped-source titlebar drag case.

**Comment by akiojin** [comment]
Progress
- Merged origin/develop into work/20260506-1747 and resolved the PR #2510 merge conflict.
- Addressed the Codex review feedback in f4b57d28.

Done
- Titlebar docking now returns no target when the source window is already in a tab group, preserving grouped-window titlebar drag as a move gesture.
- Added frontend unit coverage for the grouped-source titlebar drag case.
- Local frontend checks, Rust checks, commit hooks, and pre-push coverage checks passed.

Next
- Await refreshed GitHub Actions for the latest push.
- SPEC-2008 T-121 manual GUI docking smoke remains before the draft PR should be marked ready.

File: crates/gwt/web/window-docking.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (akiojin) [reply crates/gwt/web/window-docking.js:32]: Fixed in 009b5c1e: findTitlebarDockTarget now returns null for missing or non-finite pointer points before reading x/y. Added frontend unit coverage for null, NaN, and Infinity pointer inputs.

**Comment by akiojin** [comment]
Progress
- Addressed the follow-up CodeRabbit comments after f4b57d28.

Done
- Added a defensive pointer guard in `findTitlebarDockTarget` for null, NaN, and Infinity inputs.
- Simplified the SPEC-2008 lessons entry to a conceptual takeaway and linked details to SPEC #2008 / PR #2510.
- Added frontend unit coverage for invalid pointer inputs.
- Latest local/pre-push verification passed with coverage above threshold.

Next
- PR #2510 has been merged by auto-merge. SPEC-2008 T-121 manual GUI smoke remains as follow-up before closing the overall phase.
